drivers/crypto/crypto_mtls_shim.c

```diff
@@ -87,6 +93,11 @@ static int mtls_ccm_decrypt_auth(struct cipher_ctx *ctx,
 		return -EINVAL;
 	}
 
+	/* Only tag_len bytes are compared in mbedtls_ccm_auth_decrypt()
+	 * if the decryption was successful.
+	 */
+	apkt->pkt->out_len = ctx->mode_params.ccm_info.tag_len;
```

Comment:
Why only tag_len is returned? Shouldn't it be same as in CBC decrypt case?

Reply:
There's a comment right above that explaining the decision.  Does it make sense?

Reply:
My point is that the mtls_ccm_decrypt_auth() function is expected to return the length of decrypted text in 'out_len'. So irrespective of 'tag_len' bytes getting compared in mbedtls_ccm_auth_decrypt(), it has to be length of decrypted text. Tag_len is constant value for a given cipher.